crates/re_viewer_context/src/space_view/view_context_system.rs

```diff
@@ -1,19 +1,23 @@
 use ahash::HashMap;
 
+use re_types::ComponentNameSet;
+
 use crate::{
-    ArchetypeDefinition, NamedViewSystem, SpaceViewSystemExecutionError, ViewQuery, ViewSystemName,
-    ViewerContext,
+    NamedViewSystem, SpaceViewSystemExecutionError, ViewQuery, ViewSystemName, ViewerContext,
 };
 
 /// View context that can be used by view parts and ui methods to retrieve information about the scene as a whole.
 ///
 /// Is always populated before view part systems.
 pub trait ViewContextSystem {
-    /// Each scene context may query several archetypes.
+    /// Each scene context may _require_ several different set of components in order to be
+    /// instantiated.
+    ///
+    /// This lists all sets that the context requires.
     ///
-    /// This lists all archetypes that the context queries.
-    /// A context may also query no archetypes at all and prepare caches or viewer related data instead.
-    fn archetypes(&self) -> Vec<ArchetypeDefinition>;
+    /// A context may also not require any components at all and merely prepare caches or viewer
+    /// related data instead.
+    fn compatible_component_sets(&self) -> Vec<ComponentNameSet>;
```

Comment:
This docstring could use some work.

> Each scene context may _require_ several different set of components in order to be instantiated.

If something requires several sets, it means it requires the union of those sets. But then let's just return the union instead of a list of sets?
But I think that is not the intended meaning here. 

> This lists all sets that the context requires.

Ok - but what are those sets used for? 

> A context may also not require any components at all and merely prepare caches or viewer related data instead.

Ok - should I return a `vec![]` or a `vec![ComponentNameSet::default()]` then? What is the difference?